### PR TITLE
Benchmark `broadcast + add`

### PR DESCRIPTION
**Pull request by Priya2698**
Ports [CPP benchmark](https://github.com/NVIDIA/Fuser/blob/main/benchmarks/cpp/broadcast.cpp) to python.
The fusion definition corresponds to:
```
def bcast_add(x, bias, bcast_dim):
    return x + bias.unsqueeze(bcast_dim)
```
This benchmark uses 2D inputs. The parameters here are:
`bcast_axis`: outer or inner (0 / 1), specifies which axis in bias is broadcasted.
`contiguous`: True or False, specifies if `x` is contiguous in memory.

### CODE REVIEW

File: benchmarks/python/test_broadcast_add_fwd.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        dtype=dtype,
+        is_cpu=False,
+        stride_order=[1, 0],
+    )
+
+    bcast_shape = [T1.size(0), T1.size(1)]
+    bcast_shape[bcast_axis] = 1
+
+    T2 = fd.ops.broadcast_in_dim(T0, shape=bcast_shape, broadcast_dims=[1 - bcast_axis])
+    T3 = fd.ops.broadcast_in_dim(T2, shape=T1.shape(), broadcast_dims=[0, 1])
```
Comment (naoyam) [inline benchmarks/python/test_broadcast_add_fwd.py:32]: Why are there two broadcast ops? Is the second one necessary?

File: benchmarks/python/test_broadcast_add_fwd.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+def test_bcast_add_baseline_benchmark(
+    benchmark,
+    size: tuple,
+    dtype: torch.dtype,
+    bcast_axis: int,
+    compile: bool,
+):
+    clear_cuda_cache()
+    if compile:
+        clear_dynamo_cache()
```
Comment (naoyam) [inline benchmarks/python/test_broadcast_add_fwd.py:89]: nit: is there anything like a hook where we can register some common operations before each benchmarking? I assume we need to do these clear ops with all benchmarks. I'd expect testing/benchmarking frameworks would have some mechanism to simplify such common operations.

File: benchmarks/python/test_broadcast_add_fwd.py (hunk 1 above)
Comment (Priya2698) [reply benchmarks/python/test_broadcast_add_fwd.py:32]: No, we can combine them into a separate op.
But I kept the fusion definition consistent with what Thunder generated.

For outer broadcast, if the fn is `x + bias`, then Thunder only produces one bcast op.
The use of `unsqueeze` generates 2 broadcast ops.

File: benchmarks/python/test_broadcast_add_fwd.py (hunk 2 above)
Comment (Priya2698) [reply benchmarks/python/test_broadcast_add_fwd.py:89]: Good point, I'll look into it.

File: benchmarks/python/test_broadcast_add_fwd.py (hunk 1 above)
Comment (naoyam) [reply benchmarks/python/test_broadcast_add_fwd.py:32]: OK, can you just leave a comment about that?

**Comment by Priya2698** [comment]
CC: @xwang233 for additional parameters.

**Comment by csarofeen** [comment]
LGTM pending Naoya's comments. Thanks!

File: benchmarks/python/test_broadcast_add_fwd.py (hunk 1 above)
Comment (Priya2698) [reply benchmarks/python/test_broadcast_add_fwd.py:36]: Added a comment.

File: benchmarks/python/test_broadcast_add_fwd.py (hunk 2 above)
Comment (Priya2698) [reply benchmarks/python/test_broadcast_add_fwd.py:110]: I'll pursue this is in a separate PR, since it will be a module-wide change.

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**